client/question/services/code_evaluators/PythonPrereqCheckService.js

```diff
@@ -89,32 +89,45 @@ tie.factory('PythonPrereqCheckService', [
     var detectAndGetWrongLanguageType = function(code) {
       var obscuredCode = getObscuredCode(code);
       var rawCodeArray = code.split('\n');
+      var lowestErrorCharNumber;
+      var firstError;
 
       for (var i = 0; i < WRONG_LANGUAGE_ERRORS.python.length; i++) {
         var error = WRONG_LANGUAGE_ERRORS.python[i];
         var regexp = new RegExp(error.regExString);
 
         // Lookup character location of error
-        var firstErrorCharNumber = obscuredCode.search(regexp);
-        if (firstErrorCharNumber !== -1) {
+        var errorCharNumber = obscuredCode.search(regexp);
+
+        // Regex search returns -1 for no match. Use reverse King-of-the-hill
+        // check to find lowest number.
+        if (errorCharNumber > -1) {
+          if (firstError && errorCharNumber > lowestErrorCharNumber) {
```

Comment:
should be `>=`, perhaps

Reply:
The only way that >= would return valid for 105 would be if two wrongLang errors occur in the exact same place. If that's the case, the line number would be correct.

Reply:
Ah, I was thinking more in terms of saving time; presumably you don't need to process stuff if the char number is the same. But this is optional.